metricbeat/module/docker/docker.go

```diff
@@ -84,6 +84,7 @@ func FetchStats(client *docker.Client) ([]Stat, error) {
 	}()
 
 	wg.Wait()
+	close(queue)
```

Comment:
Consider swapping things around so that you can call `wg.Done()` from the worker goroutines. This more closely aligns with the [Fan-in, Fan-out](https://blog.golang.org/pipelines#TOC_4.) examples here.

```
 	for _, container := range containers {
 		go func(container docker.APIContainers) {
			defer wg.Done()
 			queue <- exportContainerStats(client, &container)
 		}(container)
 	}

 	go func() {
  	  	wg.Wait()
  	  	close(queue)
  	}()
  
  	// This will break after the queue and been drained and queue is closed.
 	for container := range queue {
 		containersList = append(containersList, container)
 	}

```